src/Orleans/Storage/RelationalConstants.cs

```diff
@@ -0,0 +1,379 @@
+    /// </summary>
+    public static class WellKnownRelationalInvariants
+    {
+        /// <summary>
+        /// Microsoft SQL Server.
+        /// </summary>
+        public const string SqlServer = "System.Data.SqlClient";
+
+        /// <summary>
+        /// Oracle Database server.
+        /// </summary>
+        public const string OracleDatabase = "Oracle.DataAccess.Client";
+    }
+
+
+    /// <summary>
+    /// Constants Orleans uses to relational database management.
+    /// </summary>
+    [DebuggerDisplay("InvariantName = {InvariantName}, AssemblyQualifiedName = {AssemblyQualifiedName}")]
+    public class RelationalConstants
+    {
+        /// <summary>
+        /// A key for a well known constant, likely set by a database connector vendor.
+        /// </summary>
+        public const string NameKey = "Name";
+
+        /// <summary>
+        /// A key for a well known constant, likey set by a database connector vendor.
+        /// </summary>
+        public const string DescriptionKey = "Description";
+
+        /// <summary>
+        /// A key for a well known constant, likely set by a database connector vendor.
+        /// </summary>
+        public const string InvariantNameKey = "InvariantName";
+
+        /// <summary>
+        /// A key for a well known constant, likely set by a database connector vendor.
+        /// </summary>
+        public const string AssemblyQualifiedNameKey = "AssemblyQualifiedName";
+
+        /// <summary>
+        /// A key to add a default connection string if none otherwise defined.
+        /// This is something a given database will be listening to. If not altered during
+        /// setup, usually databases have well known default settings.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string DefaultConnectionStringKey = "DefaultConnectionString";
+
+        /// <summary>
+        /// The character that indicates a parameter.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string ParameterIndicatorKey = "ParameterIndicator";
+
+        /// <summary>
+        /// The character that indicates a start escape key for columns and tables that are reserved words.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string StartEscapeIndicatorKey = "StartEscapeIndicator";
+
+        /// <summary>
+        /// The character that indicates an end escape key for columns and tables that are reserved words.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string EndEscapeIndicatorKey = "EndEscapeIndicator";
+
+        /// <summary>
+        /// A key for a query template to create a database with a given name.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string CreateDatabaseKey = "CreateDatabaseTemplate";
```

Comment:
These are used to make adding databases to tests more easily. Say, one wants to add a MySQL backend and test the membership implementation with it. Adding the needed queries to create and delete a database (used by tests) here makes it easy (should make, currently the tests are hardcoded to use SQL Server as there's no way to check provider invariant string from configs) as changing a connection string.